frmts/vrt/pixelfunctions.cpp

```diff
@@ -1471,6 +1471,46 @@ static CPLErr ScalePixelFunc(void **papoSources, int nSources, void *pData,
     return CE_None;
 }
 
+static CPLErr NormDiffPixelFunc(void **papoSources, int nSources, void *pData,
+                                int nXSize, int nYSize, GDALDataType eSrcType,
+                                GDALDataType eBufType, int nPixelSpace,
+                                int nLineSpace)
+{
+    /* ---- Init ---- */
+    if (nSources != 2)
+        return CE_Failure;
+
+    if (GDALDataTypeIsComplex(eSrcType))
+    {
+        CPLError(CE_Failure, CPLE_AppDefined,
+                 "norm_diff cannot by applied to complex data types");
+        return CE_Failure;
+    }
+
+    /* ---- Set pixels ---- */
+    size_t ii = 0;
+    for (int iLine = 0; iLine < nYSize; ++iLine)
+    {
+        for (int iCol = 0; iCol < nXSize; ++iCol, ++ii)
+        {
+            const double dfLeftVal = GetSrcVal(papoSources[0], eSrcType, ii);
+            const double dfRightVal = GetSrcVal(papoSources[1], eSrcType, ii);
+
+            const double dfPixVal =
+                (dfLeftVal - dfRightVal) / (dfLeftVal + dfRightVal);
```

Comment:
@avalentino Can do. How would you recommend I handle it? I was using the `div` function as a model, and the documentation says it doesn't check for it:

https://github.com/OSGeo/gdal/blob/5ad59397a6fce67ef991362f6c8f8f9127e7acb3/frmts/vrt/pixelfunctions.cpp#L1499

Reply:
The doc has fallen behind the implementation. There's now a check for division by zero in the inverse function:
https://github.com/OSGeo/gdal/blob/5ad59397a6fce67ef991362f6c8f8f9127e7acb3/frmts/vrt/pixelfunctions.cpp#L934
I believe this was made to please oss-fuzz that tags that as undefined behaviour otherwise. I'd say to simply go for a plain check against 0 (although I see your point about checking against some epsilon)